flink-core/src/main/java/org/apache/flink/types/parser/DoubleParser.java

```diff
@@ -103,6 +112,10 @@ public static final double parseField(byte[] bytes, int startPos, int length, ch
 		}
 		
 		String str = new String(bytes, startPos, i);
+		int len = str.length();
+		if(len > str.trim().length()) {
```

Comment:
`String.trim()` creates a new String object. 
Checking if the first or last character of the String is a whitespace is probably more efficient.